river/main.zig

```diff
@@ -42,6 +42,13 @@ const usage: []const u8 =
 pub var server: Server = undefined;
 
 pub fn main() anyerror!void {
+    const act = os.Sigaction{
+        .handler = .{ .sigaction = os.SIG.IGN },
+        .mask = os.empty_sigset,
+        .flags = os.SA.SIGINFO,
+    };
+    os.sigaction(os.SIG.PIPE, &act, null);
+
```

Comment:
If I understand the `signal(7)` man page and related man pages correctly, setting the signals disposition to ignore like this is different from using `sigprocmask` to block the signal. I think your code here is correct and we do indeed want to ignore this signal rather than block it, but that means we need to un-ignore it in a forked child process before calling `execve()`. See the two existing appearances of `sigprocmask()` in river's code.

Reply:
I just copied over the function from that zig issue actually, skimmed some docs and promptly missed this detail. now I did some more digging and I'm not sure if what the code does is correct any more (using siginfo and SIG_IGN at the same time) https://lists.gnu.org/archive/html/bug-gnulib/2008-06/msg00196.html

Reply:
So I suppose the better move as add our own handler that ignores the signal? this way it is also automatically reset when fork-exec right?

Reply:
> So I suppose the better move is add our own handler that ignores the signal? this way it is also automatically reset when fork-exec right?

Ah, yeah that seems to do what we want without requiring us to remember anything on fork.